docs(tax): DEVDOCS-2730 Added sandbox section, other minor improvements (#1052)
bigcommerce/dev-docs: docs/api-docs/provider-apis/tax/overview.md

| Merchant support email                | Optional                       | Email                        | Displayed in the BigCommerce control panel in **Store Setup > Tax > Tax Provider**.                       | `support@sampletax.example.com`                   |
| Merchant support URL                  | Optional                       | URL                          | Displayed in the BigCommerce control panel in **Store Setup > Tax > Tax Provider**.                        | `sampletax.example.com/support`                   |
| **Coverage**                          |                                |                              |                                                                                                       |                                                   |
| Tax provider visibility               | Required                       | Show, Hide                   | Tells us if your tax provider should be displayed to users in the BigCommerce control panel.          | `Show`                                            |

> | Supported store(s)                    | Required                       | All stores, Private instance | Tells us which stores your tax provider is supported on.                                              | `All stores`                                      |

Reviewer: Hmm perhaps we do need to change this title, given yesterday's discussion.

This is more like "Platform Availability". This (or similar) differentiates it better from the private instance concept below and hence likely adds a bit of clarity without having to read the description.

Author: Updated with [latest commit](dea6ca188207b7ac2eae1415d97256124461e275). Thoughts?

| **Testing**                               |                                |                              |                                                                                                       |                                                   |
| Partner test store domain(s)          | Required                       | Domain name                  | Share your test store domain(s) sow we can test prior to launching your tax provider.                 | `https://sampletax-test-store.mybigcommerce.com/` |

### Sandbox tax provider configuration

Sandbox tax provider configurations are nested within your production tax provider configuration, as such tax providers requiring a sandbox tax provider configuration should also provide the details for their production tax provider configuration.

> Additionally, when providing the details for a sandbox tax provider configuration, it's unncessary to specify the below properties because they are inherited from the parent production tax provider configuration (all other properties are still required):

Reviewer: ```suggestion
Additionally, when providing the details for a sandbox tax provider configuration, it's unnecessary to specify the below properties because they are inherited from the parent production tax provider configuration (all other properties are still required):
```

* Supported store(s), i.e. all stores or supported store hashes in the case tax provider is a private instance

> * Supported / unsupported countries

Reviewer: We also don't need: 
* Tax provider visibility
* Merchant support email  
* Merchant support URL

Author: Updated with [latest commit](dea6ca188207b7ac2eae1415d97256124461e275).

## Building the app

Tax providers are required to build a BigCommerce [single-click app](https://developer.bigcommerce.com/api-docs/apps/guide/types#single-click) in order to utilise the Tax Provider API to provide tax estimates and submit tax documents. A BigCommerce single-click app provides many benefits, for example, it enables tax providers to promote their solution in the BigCommerce apps marketplace, ask for merchant authorisation of API scopes during app install, as well as enable configuration of tax provider settings via an iFrame in the BigCommerce control panel.

Review our [introduction to building apps](https://developer.bigcommerce.com/api-docs/apps/guide/intro) guide and use the sidebar to explore topics including: types of apps, managing apps in the dev portal, implementing OAuth, and designing the UI.

Make sure to also review our [app development best practices](https://developer.bigcommerce.com/api-docs/apps/guide/best-practices) for some tips.

<div class="HubBlock--callout">
<div class="CalloutBlock--info">
<div class="HubBlock-content">

> ### Note

> > * If you have registered your app in the Developer Portal but have not submitted it for approval because it's still in development, the app will be in a Draft state. This means your app can only be installed on stores owned by the same email address as the Developer Portal account email.

Reviewer: Nice call out - great to have these items inline if we commonly come across them.

| **Testing**                               |                                |                              |                                                                                                       |                                                   |
| Partner test store domain(s)          | Required                       | Domain name                  | Share your test store domain(s) sow we can test prior to launching your tax provider.                 | `https://sampletax-test-store.mybigcommerce.com/` |

### Sandbox tax provider configuration

Sandbox tax provider configurations are nested within your production tax provider configuration, as such tax providers requiring a sandbox tax provider configuration should also provide the details for their production tax provider configuration.

Additionally, when providing the details for a sandbox tax provider configuration, it's unncessary to specify the below properties because they are inherited from the parent production tax provider configuration (all other properties are still required):

* Supported store(s)

> * Supported store hash(es) (in the case tax provider is a private instance)

Reviewer: What are the differences here?

Author: Thinking about it some more, given in the code **support_stores** is associated to store hash values and when this is blank we assume all stores are supported, the above is a bit duplicative. Would it make more sense if I remove the first line, "Supported store(s)" and leave the second?

Reviewer 2: I think so

Author: Updated [here](1e09ba04983591550d8f744afb960444d220c2bb), thoughts?

Reviewer: Lovely, thanks @bc-rogq.

Document submission enables tax providers to persist a tax quote request, replace it with another one, or invalidate it if necessary.

> Supporting document submission is optional; however, tax providers wishing to support this functionality are required to share document submission URLs with BigCommerce when [sharing provider details with BigCommerce](#sharing-provider-details-with-bigcommerce).

Reviewer 2: 💯 makes much more sense

Document submission enables tax providers to persist a tax quote request, replace it with another one, or invalidate it if necessary.

> Supporting document submission is optional; however, tax providers wishing to support this functionality are required to share document submission URLs with BigCommerce when [sharing provider details with BigCommerce](#sharing-provider-details-with-bigcommerce).

Reviewer 3: ```suggestion
Supporting document submission is optional. However, tax providers wishing to support this functionality must share document submission URLs with BigCommerce when [sharing provider details with BigCommerce](#sharing-provider-details-with-bigcommerce).

```
